torchx/tracker/mlflow.py

```diff
@@ -129,10 +152,14 @@ def get_run(self, run_name: str) -> Run:
             )
             if not search_result:
                 return mlflow.start_run(
-                    experiment_id=self.experiment_id, run_name=run_name
+                    experiment_id=self.experiment_id,
+                    run_name=run_name,
                 )
             elif len(search_result) == 1:
-                return search_result[0]
+                return mlflow.start_run(
+                    experiment_id=self.experiment_id,
+                    run_id=search_result[0].info.run_id,
+                )
             else:  # len(search_result) > 1
                 raise RuntimeError(
```

Comment:
what's the risk of this race condition? should we always do a second check after calling start_run and deleting the one with the larger run_id?

Reply:
the race is actually handled in the ctor of `MLflowTracker` where we wrap the create-or-query experiment/run `with on_rank0_first():`. The issue that this PR is solving is that when the non-zero ranks query the run, we were not **setting** the active run with the queried run_id which means that we never end up logging artifacts, metrics, etc against the job's run_id.